### PR TITLE
Add WalletConnect Modal support

### PR DESCRIPTION
**Pull request by ruixhuang**
Also updated the sample app to show the usage:

https://github.com/user-attachments/assets/e4631ac9-3815-4076-8902-4e0777610cac

### CODE REVIEW

**Review by tinaszheng — DISMISSED** [review]

File: app/src/main/java/exchange/dydx/carteraExample/WalletListViewModel.kt [hunk 1]
```diff
@@ -101,15 +121,21 @@ class WalletListViewModel(
...
+        val request = WalletRequest(wallet = wallet, address = null, chainId = chainId, context = context, useModal = useModal)
         provider.signMessage(
             request = request,
             message = "Test Message",
             connected = { info ->
                 Log.d(tag(this@WalletListViewModel), "Connected to: ${info?.peerName ?: info?.address}")
             },
+            status = { requireAppSwitching ->
+                Log.d(tag(this@WalletListViewModel), "Require app switching: $requireAppSwitching")
+                toastMessage("Please switch to the wallet app")
```
[the remark this answers is not in the document]
Comment (ruixhuang) [reply app/src/main/java/exchange/dydx/carteraExample/WalletListViewModel.kt:134]: This is for the test app only, and won't be part of the production code.  I will add proper localized text to the production mobile app code.

File: cartera/src/main/java/exchange/dydx/cartera/entities/ModelExtensions.kt [hunk 2]
```diff
@@ -51,3 +56,24 @@ fun WalletConfig.connectionType(context: Context): WalletConnectionType {
 fun WalletConfig.connections(type: WalletConnectionType): WalletConnections? {
     return connections?.firstOrNull { it.type == type.rawValue }
 }
+
+fun EthereumTransactionRequest.toJsonRequest(): String? {
+    val request: MutableMap<String, Any?> = mutableMapOf()
+
+    request["from"] = fromAddress
+    request["to"] = toAddress ?: "0x"
+    request["gas"] = gasLimit?.toHexString()
```
[the remark this answers is not in the document]
Comment (ruixhuang) [reply cartera/src/main/java/exchange/dydx/cartera/entities/ModelExtensions.kt:65]: We can keep this here for backward compatibility.  Those are optional fields and client can decided whether to use it.

File: app/src/main/java/exchange/dydx/carteraExample/WalletListViewModel.kt [hunk 3]
```diff
@@ -101,15 +121,21 @@ class WalletListViewModel(
...
+            },
             completion = { signature, error ->
+                // delay for 1 second
```
[the remark this answers is not in the document]
Comment (ruixhuang) [reply app/src/main/java/exchange/dydx/carteraExample/WalletListViewModel.kt:137]: Updated

**Review by jaredvu — APPROVED** [review]
@tinaszheng stamping for now as it is blocking mobile. feel free to retro-actively review the PR

**Pull request merged**